test/register.test.js

```diff
@@ -128,31 +128,23 @@ test('awaitable register and after', async t => {
 
 test('awaitable after error handling', async t => {
   const fastify = Fastify()
-  let first = false
-  let second = false
-  let third = false
 
-  // Note that this does not throw in case of error,
-  // all errors are currently collected to ready() because it's
-  // not really possible to do anything about them.
-  await fastify.register(async (instance, opts, next) => {
-    first = true
-    throw new Error('kaboom')
+  const e = new Error('kaboom')
+
+  // TODO this should throw for some reason, and it's not.
+  await fastify.register(async (instance, opts) => {
+    throw e
   })
-  t.is(first, true)
 
-  fastify.register(async (instance, opts, next) => {
-    second = true
+  fastify.register(async (instance, opts) => {
+    t.fail('should not be executed')
   })
 
-  await fastify.after()
-  // TODO this is a bug, it should be false
-  t.is(second, true)
+  await t.rejects(fastify.after(), e)
```

Comment:
after should not throw https://github.com/mcollina/avvio/blob/fba94f7cd1572cf400d38bcd64a7c37bf2a46f46/test/await-after.test.js#L152

```suggestion
  await fastify.after()
```